### PR TITLE
Make cross op structured

### PR DESCRIPTION
**Pull request by ramiro050**
Summary:
This commit makes the cross op into a structured kernel.

@bdhirsh, I'm new to structured kernels, so I would love any feedback on this!

### CODE REVIEW

File: aten/src/ATen/native/Cross.cpp [hunk 1]
```diff
@@ -48,7 +52,6 @@ Tensor & cross_out(const Tensor & input, const Tensor & other, const c10::option
   }
```
[the remark this answers is not in the document]
Comment (mruberry) [reply aten/src/ATen/native/Cross.cpp:52]: Most of our operators today will unfortunately resize incorrectly sized out tensors. The `resize_output()` function

https://github.com/pytorch/pytorch/blob/a6c0edff1a60e299d3f9794407f0a1c9a269fd19/aten/src/ATen/native/Resize.h#L24

will throw a warning. Once all our operators are updated to use it when we can switch that warning to an error. There is an OpInfo test that verifies the warning is thrown, too, so we'll know we've caught every instance of the problem.

[the remark this answers is not in the document]
Comment (mruberry) [reply aten/src/ATen/native/Cross.cpp:52]: Follow-up: if an out tensor is the correct shape it should not be unconditionally resized and that behavior can be treated as a bug. Instead its shape should be respected.

**Comment by AnirudhDagar** [comment]
Please wait for #63285 (about to be merged). Also, #64355 is already up by @krshrimali to port cross to structured, you may want to sync up.

Ps. I strongly feel we need to keep the structured kernel tracker #55070 updated (even with potential open PRs) otherwise we end up implementing twice and with conflicting work.

**Comment by bdhirsh** [comment]
Ah, sorry for the churn @ramiro050. I missed that cross PR when I was making the list of ops.

**Comment by bdhirsh** [comment]
> Ps. I strongly feel we need to keep the structured kernel tracker #55070 updated (even with potential open PRs) otherwise we end up implementing twice and with conflicting work.

Agreed, I haven't had time to get to updating it. The silver lining is that it's pretty easy to tell if an op is already structured directly in source code (by opening `native_functions.yaml` and looking for a `structured*` keyword in the op's entry). And for open PR's that are actively porting, the issue already recommends that you should check any open PR's for sanity.

**Pull request closed without merging**